src/test/java/org/opensearch/sdk/TestExtensionRestApi.java

```diff
@@ -0,0 +1,44 @@
+package org.opensearch.sdk;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.opensearch.test.OpenSearchTestCase;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+public class TestExtensionRestApi extends OpenSearchTestCase {
+
+    private ExtensionRestPaths extensionRestApi;
+
+    @Override
+    @BeforeEach
+    public void setUp() throws Exception {
+        super.setUp();
+        File file = new File(ExtensionRestPaths.class.getResource(ExtensionRestPaths.EXTENSION_REST_PATHS_DESCRIPTOR).getPath());
+        ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
+        extensionRestApi = objectMapper.readValue(file, ExtensionRestPaths.class);
+    }
+
+    @Test
```

Comment:
I see we have covered the happy path. Any failure cases?

Reply:
Not sure what you are asking here. Either we correctly read the three `String`s from the yml, or it fails and we get an `IOException` in the `setup()` above or one of the assertions below fails.  I see no case where the below test succeeds and there is a problem in the class.